iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs

```diff
@@ -58,21 +41,21 @@ public async Task InitLoopAsync(TimeSpan timeout)
                     ConnectionString,
                     ConnectionString.AmqpEndpoint,
                     Audience,
-                    Audience,
-                    AccessRightsHelper.AccessRightsToStringArray(AccessRights.DeviceConnect),
-                    timeoutHelper.RemainingTime()
+                    WebUtility.UrlEncode(Audience),
```

Comment:
@varunpuranik @davilu I doubt this is correct - we need to Uri (System.Uri) encode which is different what WebUtility does (some form of percent encoding).

Reply:
BuildAudience is already doing a UrlEncode of the Audience - this would do this twice, right? 
We need to make sure all code paths that get the Audience are consistent (including the non-connection string ones like the Edge scenario)

Reply:
This parameter is not used in the code. It is used in `CbsLink` to call `ICbsTokenProvider::GetTokenAsync(Uri namespaceAddress, string appliesTo, string[] requiredClaims)` as parameter `appliesTo`. It is ignore in SDK code. Please refer to `IotHubConnectionString` and `IotHubConnectionString.core`.